src/hooks/useTasksProvider.tsx

```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+import { TasksContext } from 'providers';
+
+export const useTasksProvider = () => React.useContext(TasksContext);
```

Comment:
Обычно в кастомных хуках опускается Provider или Context, отчасти это для того чтобы скрыть логику получения данных сверху, не факт что они приходят из провайдера

```suggestion
export const useTasks = () => React.useContext(TasksContext);
```